internal/services/codesigning/trusted_signing_account_resource.go

```diff
@@ -0,0 +1,245 @@
+			if len(model.Sku) > 0 {
+				ele := model.Sku[0]
+				req.Properties.Sku = pointer.To(codesigningaccounts.AccountSku{})
+				req.Properties.Sku.Name = codesigningaccounts.SkuName(ele.Name)
+			}
+			req.Tags = &model.Tags
+
+			future, err := client.Create(ctx, id, req)
+			if err != nil {
+				return fmt.Errorf("creating %s: %v", id, err)
+			}
+
+			if err := future.Poller.PollUntilDone(ctx); err != nil {
+				return fmt.Errorf("waiting for creation of %s: %v", id, err)
+			}
+
+			meta.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (m TrustedSigningAccountResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			id, err := codesigningaccounts.ParseCodeSigningAccountID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			client := meta.Client.CodeSigning.V20240930previewClient.CodeSigningAccounts
+			result, err := client.Get(ctx, *id)
+			if err != nil {
+				return err
+			}
+
+			if result.Model == nil {
+				return fmt.Errorf("retrieving %s got nil model", id)
+			}
+			var output TruestedSigningAccountModel
+			output.Name = id.CodeSigningAccountName
+			output.ResourceGroupName = id.ResourceGroupName
+
+			if result.Model == nil {
+				return fmt.Errorf("Get response nil Model")
+			}
+			model := result.Model
+			output.Location = model.Location
+			output.Tags = pointer.From(model.Tags)
+			if ptrProp := model.Properties; ptrProp != nil {
+				itemProp := *ptrProp
+				output.AccountUri = pointer.From(itemProp.AccountUri)
+				if ptrSku := itemProp.Sku; ptrSku != nil {
+					itemSku := *ptrSku
+					var accountSku Sku
+					accountSku.Name = string(itemSku.Name)
+					output.Sku = append(output.Sku, accountSku)
+				}
+			}
+
+			return meta.Encode(&output)
+		},
+	}
+}
+
+func (m TrustedSigningAccountResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) (err error) {
+			client := meta.Client.CodeSigning.V20240930previewClient.CodeSigningAccounts
+			id, err := codesigningaccounts.ParseCodeSigningAccountID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			var model TruestedSigningAccountModel
+			if err = meta.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %s: %+v", id, err)
+			}
+
+			var upd codesigningaccounts.CodeSigningAccountPatch
+
+			if meta.ResourceData.HasChange("sku") {
+				upd.Properties = pointer.To(codesigningaccounts.CodeSigningAccountPatchProperties{})
+				if len(model.Sku) > 0 {
+					ele := model.Sku[0]
+					upd.Properties.Sku = pointer.To(codesigningaccounts.AccountSkuPatch{})
+					upd.Properties.Sku.Name = pointer.To(codesigningaccounts.SkuName(ele.Name))
+				}
+			}
+			if meta.ResourceData.HasChange("tags") {
+				upd.Tags = pointer.To(model.Tags)
+			}
+
+			if _, err = client.Update(ctx, *id, upd); err != nil {
```

Comment:
Can we use the get-then-put pattern instead?

Reply:
the pattern might not fit since the `client.Update()` requires a `xxxPatch{}` object, the model received from `client.Get()` is `xxxModel{}`, which can only be used with `client.Create()`. While when calling `client.Create()` towards an existing account, it errored 
```
performing Create: unexpected status 400 (400 Bad Request) with error: BadResourceOperation: The resource 'acctest-f0io7' already exists. A resource with the same name cannot be created. Please select a new resource name.
```